src/messages/mod.rs

```diff
@@ -43,17 +45,35 @@ pub enum Message {
     Hop(SignedRoutingMessage),
 }
 
+#[derive(Debug, Eq, PartialEq, Hash, Deserialize)]
+#[allow(clippy::large_enum_variant)]
+pub enum MessagePeek {
+    /// A message sent between two nodes directly
+    Direct(SignedDirectMessage),
+    /// A message sent across the network (in transit)
+    Hop(PeekedSignedRoutingMessageInfo),
+}
+
 #[allow(clippy::large_enum_variant)]
 pub enum MessageWithBytes {
     Hop(HopMessageWithBytes),
     Direct(SignedDirectMessage, Bytes),
 }
 
 impl MessageWithBytes {
-    pub fn from_bytes(bytes: Bytes) -> Result<Self> {
-        match from_network_bytes(&bytes)? {
-            Message::Hop(msg) => Ok(Self::Hop(HopMessageWithBytes::new_from_parts(msg, bytes))),
-            Message::Direct(msg) => Ok(Self::Direct(msg, bytes)),
+    pub fn peek_from_bytes(bytes: Bytes) -> Result<Self> {
+        match peek_from_network_bytes(&bytes)? {
+            MessagePeek::Hop(msg_peek) => Ok(Self::Hop(HopMessageWithBytes::new_from_parts(
+                None, msg_peek, bytes,
+            ))),
+            MessagePeek::Direct(msg) => Ok(Self::Direct(msg, bytes)),
+        }
+    }
+
+    pub fn full_hop_msg_from_bytes(bytes: &Bytes) -> Result<SignedRoutingMessage> {
+        match from_network_bytes(bytes)? {
```

Comment:
the `from_network_bytes` function could be used to handle a Direct message, but here we only expect it to handle hop message?

Reply:
yes, it is a full *hop* message from bytes. The direct message was already deserialized on its own.